pkg/dataobj/consumer/partition_processor.go

```diff
@@ -54,14 +58,22 @@ func newPartitionProcessor(ctx context.Context, client *kgo.Client, builderCfg d
 	}
 	reg = prometheus.WrapRegistererWith(prometheus.Labels{
 		"partition": strconv.Itoa(int(partition)),
+		"tenant":    tenantID,
 	}, reg)
 
 	metrics := newPartitionOffsetMetrics()
 	if err := metrics.register(reg); err != nil {
 		level.Error(logger).Log("msg", "failed to register partition metrics", "err", err)
 	}
 
-	metastoreManager, err := metastore.NewMetastoreManager(bucket, tenantID, logger, reg)
+	uploader := uploader.New(uploader.UploaderConfig{
+		SHAPrefixSize: builderCfg.SHAPrefixSize,
```

Comment:
I'm not sure if SHAPrefixSize should be in the builderCfg anymore. Removing it means we'd have to pass a separate uploader config to this processor.

Reply:
I think removing the SHA prefix length from the builder config makes sense, FWIW. 

If we move it out, we could put it in a upload-specific config (write only) or in a more general storage config (read + write?), whichever we think makes sense, depending on if we expect there to be overlap.